lib/puppet/indirector/direct_file_server.rb

```diff
@@ -7,9 +7,7 @@ class Puppet::Indirector::DirectFileServer < Puppet::Indirector::Terminus
 
   def find(request)
     return nil unless Puppet::FileSystem.exist?(request.key)
-    instance = model.new(request.key)
-    instance.links = request.options[:links] if request.options[:links]
-    instance
+    path2instance(request, request.key)
```

Comment:
The resolution after talking with @Iristyle and @joshcooper this afternoon was two things:
* It's not clear whether ignore_source_permissions should be passed to #collect in path2instances; it wasn't in the past, but that could still be a bug. Tracking in PUP-3906.
* The collect method is only defined for file_serving/content.rb and file_serving/metadata.rb, and now only used by path2instance. It was previously called for this code path by file_metadata/file.rb (corresponding to the metadata.rb model) and not called by file_content/file.rb (the content.rb model). The return value of file_serving/content.rb's #collect was never used, so I've removed the implementation and combined call paths to simplify the DirectFileServer and FileServer logic.